Updated contributing file with project structure (#306)
andregasser/bigbone: CONTRIBUTING.md

Please also note that the project requires at least Java 8 in order to build successfully. But we don't think this is really an issue nowadays.

### Project Structure Overview

The project comprises five primary Gradle modules:

- **bigbone:** This module contains the core code for interacting with the Mastodon REST API, including tests.

> - **bigbone-rx:** This module serves as the proxy layer for forwarding requests to the main bigbone Gradle module.

Reviewer: While it isn’t wrong that the Rx module “serves as the proxy layer”, I wouldn’t have worded it like this.

Maybe something like this?

```suggestion
- **bigbone-rx:** This module exposes methods available in the `bigbone` module as RxJava3 types so that you can use this library easily in an RxJava context.
```

Reviewer 2: sorry, my bad. @Yordaniss `bigbone-rx` is actually calling the methods in the `bigbone` module (calls are forwarded 1:1), `bigbone-rx` thereof is sort of a thin layer on top of the `bigbone` and provides reactive streams functionality. Could you adopt the wording a bit into this direction? Thank you very much!

Author: @andregasser no problem. I would use than the suggestion from @PattaFeuFeu and combine this one with your feedback 🙂

> - **bigbone-rx:** This module serves as a thin layer built on top of the "bigbone" module. It exposes methods available in the "bigbone" module as RxJava3 types, facilitating the use of this library within an RxJava context. Additionally, "bigbone-rx" enhances the functionality of "bigbone" by providing support for reactive streams.

Reviewer: > It exposes methods available in the "bigbone" module as RxJava3 types, facilitating the use of this library within an RxJava context. **Additionally**, "bigbone-rx" enhances the functionality of "bigbone" by providing support for reactive streams.

This feels like a duplication. By exposing methods as RxJava3 types we’re _already_ saying that we provide support for reactive streams

```suggestion
- **bigbone-rx:** This module serves as a thin layer built on top of the `bigbone` module. It exposes methods available in the `bigbone` module as RxJava3 types, allowing the use of this library within an RxJava context.
```

Author: thanks for feedback. I did the changes, that you suggested 🙂
